src/randomizer.py

```diff
@@ -1942,35 +1942,44 @@ def perm_chain_assignment(
     fn: ca.FuncDef, ast: ca.FileAST, indices: Indices, region: Region, random: Random
 ) -> None:
     """Combine two consecutive assignments into one chain assignment."""
-    cands: List[Tuple[int, Block]] = []
+    cands: List[Tuple[int, int, Block]] = []
 
     def rec(block: Block) -> None:
         statements = ast_util.get_block_stmts(block, False)
-        for i, (stmt, next_stmt) in enumerate(zip(statements, statements[1:])):
-            if (
-                region.contains_node(stmt)
-                and region.contains_node(next_stmt)
-                and isinstance(stmt, ca.Assignment)
-                and isinstance(next_stmt, ca.Assignment)
-                and (
-                    ast_util.equal_ast(stmt.lvalue, next_stmt.rvalue)
-                    or ast_util.equal_ast(stmt.rvalue, next_stmt.rvalue)
-                )
-                and not ast_util.is_effectful(next_stmt.rvalue)
-            ):
-                cands.append((i, block))
+
+        # Chance to create the chain assignment using a non conesecutive statement
+        if random_bool(random, 0.5):
+            shift_range = 1
+        else:
+            shift_range = random.randint(1, 5)
+
+        for shift in range(1, shift_range + 1):
+            zipped_stmts = zip(statements, statements[shift:])
+            for i, (stmt, next_stmt) in enumerate(zipped_stmts):
+                if (
+                    region.contains_node(stmt)
+                    and region.contains_node(next_stmt)
+                    and isinstance(stmt, ca.Assignment)
+                    and isinstance(next_stmt, ca.Assignment)
+                    and (
+                        ast_util.equal_ast(stmt.lvalue, next_stmt.rvalue)
+                        or ast_util.equal_ast(stmt.rvalue, next_stmt.rvalue)
+                    )
+                    and not ast_util.is_effectful(next_stmt.rvalue)
+                ):
+                    cands.append((i, shift, block))
```

Comment:
This loops reads a bit funny now. I would suggest rewriting it as something like
```py
for i, stmt in enumerate(statements):
    if not isinstance(stmt, ca.Assignment) or not region.contains_node(stmt) or ast_util.is_effectful(stmt.rvalue):
        continue
    for j, next_stmt in enumerate(statements[i+1:i+5], i+1):
        if ...:
            cands.append((i, j, block))